pkg/skaffold/jib/jib.go

```diff
@@ -30,14 +35,34 @@ func getDependencies(cmd *exec.Cmd) ([]string, error) {
 	}
 
 	// Parses stdout for the dependencies, one per line
-	// TODO(coollog) directories should be expanded recursively
 	lines := strings.Split(string(stdout), "\n")
 	var deps []string
-	for _, l := range lines {
-		if l == "" {
+	for _, dep := range lines {
+		if dep == "" {
+			continue
+		}
+
+		// Resolves directories recursively.
+		info, err := os.Stat(dep)
+		if err != nil {
+			if os.IsNotExist(err) {
+				logrus.Debugf("could not stat dependency: %s", err)
+				continue // Ignore files that don't exist
+			}
+			return nil, errors.Wrapf(err, "unable to stat file %s", dep)
+		}
+		if info.IsDir() {
+			err := filepath.Walk(dep, func(path string, info os.FileInfo, err error) error {
```

Comment:
`godirwalk.Walk` is much faster at walking directories (like 10x faster), which is important since this is going to run every second in watch mode.
It's ok to leave that this way but we should take note to improve it in another PR.